src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java

```diff
@@ -0,0 +1,148 @@
+
+/**
+ * JUnit Rule that before executing a test it starts a recording current screen
+ * and after the test is executed, it stops recording.
+ */
+public class TestRecorderRule extends TestWatcher {
+
+
+    private static final int MAX_RECORDING_TIME_SECS = 120000;
+    private static final int FRAME_RATE_PER_SEC = 60;
+    private static final int BIT_DEPTH = 16;
+    private static final float QUALITY_RATIO = 0.97f;
+
+    private JUnitScreenRecorder screenRecorder;
+
+    @Override
+    protected void starting(Description description) {
+        if(!isRecorderDisabled()) {
+            startRecording(description);
+        }
+    }
+
+    private void startRecording(Description des) {
+        GraphicsConfiguration gc = GraphicsEnvironment
+                .getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice()
+                .getDefaultConfiguration();
+
+        File movieFolder = new File("target");
+
+        String mimeType = FormatKeys.MIME_QUICKTIME;
+        String videoFormatName = VideoFormatKeys.ENCODING_QUICKTIME_ANIMATION;
+        String compressorName = VideoFormatKeys.COMPRESSOR_NAME_QUICKTIME_ANIMATION;
+        Dimension outputDimension = gc.getBounds().getSize();
+        int bitDepth = BIT_DEPTH;
+        float quality = QUALITY_RATIO;
+        int screenRate = FRAME_RATE_PER_SEC;
+        long maxRecordingTime = MAX_RECORDING_TIME_SECS;
+
+        Format outputFormatForScreenCapture = getOutputFormatForScreenCapture(videoFormatName,
+                compressorName, outputDimension,
+                bitDepth, quality, screenRate);
+
+        try {
+            this.screenRecorder = new JUnitScreenRecorder
+                    (gc, gc.getBounds(), getFileFormat(mimeType),
+                            outputFormatForScreenCapture, null, null, movieFolder, des);
+            this.screenRecorder.start();
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Exception starting test recording.", e);
+        } catch (AWTException e) {
+            throw new IllegalArgumentException("Exception starting test recording.", e);
+        }
+    }
+
+    @Override
+    protected void succeeded(Description description) {
+        if (this.screenRecorder != null) {
+            stopRecording();
+            if(!saveAllExecutions()) {
+                this.screenRecorder.removeMovieFile();
+            }
+        }
+    }
+
+    @Override
+    protected void finished(Description description) {
+        stopRecording();
+    }
+
+    private boolean isRecorderDisabled() {
+        return Boolean.getBoolean("RECORDER_DISABLED");
+    }
+
+    private boolean saveAllExecutions() {
+        return Boolean.getBoolean("RECORDER_SAVE_ALL");
+    }
+
+    private void stopRecording() {
+        if (this.screenRecorder != null && this.screenRecorder.getState() == ScreenRecorder.State.RECORDING) {
+            try {
+                TimeUnit.SECONDS.sleep(1);
```

Comment:
kind of with @oleg on this this would add 506s to the cpu time taken to run the tests. (or over 1 minute in wall time).
when recording at 60fps you should only need a max of 1/60 of a second to make sure the last relevant capture has been made - which then gets the overall time added down to a managable ballpark of 1 second.